bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/CommunicationManager.java

```diff
@@ -43,144 +58,223 @@
-                        if (ThingHandlerHelper.isHandlerInitialized(thing)) {
-                            logger.debug("Delegating update '{}' for item '{}' to handler for channel '{}'", newState,
-                                    itemName, channelUID);
-                            try {
-                                SafeMethodCaller.call(new SafeMethodCaller.ActionWithException<Void>() {
-                                    @Override
-                                    public Void call() throws Exception {
-                                        if (newState != null) {
-                                            handler.handleUpdate(channelUID, newState);
-                                        } else {
-                                            throw new IllegalStateException(
-                                                    "Trying to set state to null on channel " + channelUID);
-                                        }
-                                        return null;
-                                    }
-                                });
-                            } catch (TimeoutException ex) {
-                                logger.warn("Handler for thing {} takes more than {}ms for handling an update",
-                                        handler.getThing().getUID(), SafeMethodCaller.DEFAULT_TIMEOUT);
-                            } catch (Exception ex) {
-                                logger.error("Exception occurred while calling handler: {}", ex.getMessage(), ex);
-                            }
-                        } else {
-                            logger.debug("Not delegating update '{}' for item '{}' to handler for channel '{}', "
-                                    + "because handler is not initialized (thing must be in status UNKNOWN, ONLINE or OFFLINE).",
-                                    newState, itemName, channelUID);
-                        }
-                    } else {
-                        logger.warn("Cannot delegate update '{}' for item '{}' to handler for channel '{}', "
-                                + "because no handler is assigned. Maybe the binding is not installed or not "
-                                + "propertly initialized.", newState, itemName, channelUID);
-                    }
-                } else {
-                    logger.warn(
-                            "Cannot delegate update '{}' for item '{}' to handler for channel '{}', "
-                                    + "because no thing with the UID '{}' could be found.",
-                            newState, itemName, channelUID, channelUID.getThingUID());
+        final Item item = itemRegistry.get(itemName);
+
+        itemChannelLinkRegistry.stream().filter(link -> {
+            // all links for the item
+            return link.getItemName().equals(itemName);
+        }).filter(link -> {
+            // make sure a command event is not sent back to its source
```

Comment:
"command" or "update" event?
Or is it about the default slave profile? If so, could you add it in the comment?